proposals/0047-last-hard-fork-syscall.md

```diff
@@ -0,0 +1,120 @@
+---
+simd: '0047'
+title: Syscall to get the last hardfork
+authors:
+  - Godmode Galactus (Mango Markets)
+category: Standard
+type: Core
+status: Draft
+created: 2023-04-15
+---
+
+## Summary
+
+Create a new sysvar `SysvarLastHardFork1111111111111111111111111`, which can
+be used to get the last hard fork slot.
+
+## Motivation
+
+In Solana, when the cluster cannot reach consensus, it is restarted using
+`ledger-tool` with a hard fork on a slot. A hard fork is usually done on an
+optimistically confirmed slot which was voted on and accepted by 50% of the
+cluster but still was unable to get a supermajority. The cluster is then
+restarted from the hard fork slot, and all participating nodes need to restart
+their validator specifying the hard fork slot. Once all nodes participating in
+the restart effort on the hard fork exceed 80% of the total stakes, the restart
+is successful, and the cluster continues from the hard fork slot. So hard fork
+is a tool to intentionally fork off the nodes not participating in the restart
+effort. After successfully restarting the network the hard fork slot is stored
+in a variable called `hard_forks` in bank.
```

Comment:
"After successfully restarting the network the hard fork slot is stored
in a variable called `hard_forks` in bank." -- remove. This is an implementation detail for the Solana Labs validator client, and may not be true for other clients

Reply:
I have removed the variable name, but kept the implementation details. I specified that they are solana validator client.

I can remove that too but I do not see what I could write over here.

Reply:
outdated